src/http.c

```diff
@@ -479,10 +509,52 @@ http_stream_setup(char **stream, const char *url)
   /* Read the playlist until the first stream link is found, but give up if
    * nothing is found in the first 10 lines
    */
+  in_playlist = false;
   n = 0;
   while ((line = evbuffer_readln(ctx.input_body, NULL, EVBUFFER_EOL_ANY)) && (n < 10))
     {
+      // Skip comments and blank lines without counting for the limit
+      if (pl_format == PLAYLIST_M3U && (line[0] == '#' || line[0] == '\0'))
+        goto line_done;
+
       n++;
+
+      if (pl_format == PLAYLIST_PLS && !in_playlist)
+        {
+          if (strncasecmp(line, "[playlist]", strlen("[playlist]")) == 0)
+            {
+              in_playlist = true;
+              n = 0;
+            }
+          goto line_done;
```

Comment:
goto's are generally only for errors in forked-daapd, but I agree on your sentiment that the evilness of goto's is exaggerated, so I think it will be ok here